client/src/components/QuestionsAnswers.jsx

```diff
@@ -9,30 +10,36 @@ export default function QuestionsAnswers({ productID }) {
   const tempId = 40355;
   const [questionList, setQuestionList] = useState([]);
   const [filteredList, setFilteredList] = useState([]);
+  const [show, setShow] = useState(false);
 
-  useEffect(() => {
+  const updateQuestions = () => {
+    console.log('invoked');
     axios.get(`/api/qa/questions?product_id=${tempId}&count=5&page=1`)
       .then((result) => {
         setQuestionList(result.data.results);
       })
       .catch((err) => {
         console.log(err);
       });
+  };
+  useEffect(() => {
+    updateQuestions();
   }, [tempId]);
 
   useEffect(() => {
     setFilteredList(questionList);
   }, [questionList]);
 
   return (
-    <div id="qna">
-      Hello from QnA
+    <div className="qna">
+      <h3>Questions & Answers</h3>
       <QuestionsSearch
         questionList={questionList}
         setFilteredList={setFilteredList}
       />
-      <QuestionList questionList={filteredList} />
-      <AskQuestion />
+      <QuestionList questionList={filteredList} updateQuestions={updateQuestions} />
+      <button type="button" onClick={() => { setShow(true); }}> Ask a question</button>
+      <AskQuestion show={show} setShow={setShow} productID={tempId} update={updateQuestions} />
```

Comment:
on line 42, how about using conditional statement for this AskQuestion modal? like {show && <AskQustion .../>

Reply:
done 👍